jaxb/src/main/java/feign/jaxb/JAXBDecoder.java

```diff
@@ -66,7 +66,7 @@ private JAXBDecoder(Builder builder) {
 
   @Override
   public Object decode(Response response, Type type) throws IOException {
-    if (response.status() == 404)
+    if (response.status() == 404 || response.status() == 204)
```

Comment:
Sounds fair.

Could you just add a test for `204` as well?

Reply:
Or better yet, we should inspect the body before sending it to the `Decoder` here:

https://github.com/OpenFeign/feign/blob/d7cc9b6c7ee34da15ee2cf0e1c0824b3040e5737/core/src/main/java/feign/SynchronousMethodHandler.java#L162-L170

Reply:
I prefer a global way to handle it.

Reply:
> Should we do this globally or just in this decoder?

@kdavisk6 we could, but that would break current API for all decoders...

If you think of a `OptionalDecoder` the result for a 204 or 404 is `Optional.empty()`

For a `JAXB` decoder null
For a `List` decoder may be `Collections.emptyList()` (which I currently use for a binary format decoder.

So, although I would like to see this in a single location, I don't think we can.

Reply:
Fair enough.  Let's go with this for now and keep an eye out for other issues.